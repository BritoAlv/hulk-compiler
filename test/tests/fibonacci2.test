function fibonacci(n : number) : number => 
let index = 0, next = 1, current = 1, temp = next, condition = true in 
    while(condition)
        if (index == n)
        {
            condition := false;
            current;
        }
        else
        {
            index := index + 1;
            temp := next;
            next := next + current;
            current := temp;
        };

print(fibonacci(40));