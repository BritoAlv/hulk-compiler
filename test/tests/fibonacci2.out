165580128.00000000

